src/Storages/MergeTree/ParallelReplicasReadingCoordinator.cpp

```diff
@@ -394,12 +421,22 @@ void DefaultCoordinator::initializeReadingState(InitialAllRangesAnnouncement ann
                 throw Exception(ErrorCodes::LOGICAL_ERROR, "Intersecting parts found in announcement");
 
             known_parts.emplace(Part{.description = part, .replicas = {}});
-            all_parts_to_read.push_back(Part{.description = std::move(part), .replicas = {announcement.replica_num}});
+
+            /// Generate unique hash for each part info
+            UInt64 hash = sipHash64(part.info.getPartNameV1());
```

Comment:
If we expect no collisions, we should use 128bit.

Reply:
The number of parts is unlikely to exceed a few thousand, so the probability of collisions here is low. If we switch to 128bit, it would increase memory usage.